modules/instance_template/variables.tf

```diff
@@ -105,22 +105,32 @@ variable "disk_labels" {
   default     = {}
 }
 
+variable "disk_encryption_key" {
```

Comment:
How common/likely is it to use *different* encryption keys for the boot disk and other disks? I'm worried about the confusion of having this here _and_ under additional_disks.

Reply:
I wouldn't presume to know how common/likely it is...everyone's needs are different. 

If you're concerned about confusion, we could just keep the top level variable and have that be the key for all disks associated with the instance.  We'd likely still need an optional (defaulting to false) boolean for each additional disk to indicate then whether or not the disk should be encrypted or not using the top level defined encryption key.  That in itself seems to me more likely to cause confusion than the currently proposed approch.

Unless of course you want to force an "all disks get encrypted with the same key, or no disk encryption at all" rule.

Reply:
What do you need for your customer scenario?

In general, we try to solve for 90% of cases instead of making the module interface overly complicated to solve 100% of them.

My preference is to force "all disks get encrypted with the same key, or no disk encryption at all" because I suspect in the vast majority of cases the same key is used for all disks.